src/Compilers/Core/Portable/DocumentationCommentId.cs

```diff
@@ -70,7 +77,15 @@ public static string CreateReferenceId(ISymbol symbol)
 
             if (symbol is INamespaceSymbol)
             {
-                return CreateDeclarationId(symbol);
+                // This is very odd.  For anything other than a namespace, we defer to ReferenceGenerator, which just
+                // spits out a reference.  But for a namespace, we spit out a declaration (which means we prefix with
+                // `N:`).  None of the other paths do this.  This is likely a bug, but it appears as if it has never
+                // been noticed.
```

Comment:
Something super odd i noticed.  This is in CreateReferenceId (which does not append the `X:` prefix on any symbol).  But for namespaces... it does.  This can't possibly be expected.  My guess is that no one actually ever tried a namespace symbol here (or tested it).

Reply:
Note: i'm fine changing this.  But it was out of scope for the problem i was trying to address, so i left it alone.  However, it very much was found to be a problem thanks to the approach to auditing taken in this pr.